src/main/java/com/kobylynskyi/graphql/codegen/mapper/InterfaceDefinitionToDataModelMapper.java

```diff
@@ -44,15 +40,21 @@ public InterfaceDefinitionToDataModelMapper(GraphQLTypeMapper graphQLTypeMapper,
     public Map<String, Object> map(MappingContext mappingContext, ExtendedInterfaceTypeDefinition definition) {
         Map<String, Object> dataModel = new HashMap<>();
         // type/enum/input/interface/union classes do not require any imports
+        String clazzName  = dataModelMapper.getModelClassNameWithPrefixAndSuffix(mappingContext, definition);
+        List<ParameterDefinition> fields = fieldDefinitionToParameterMapper.mapFields(mappingContext, definition.getFieldDefinitions(), definition);
         dataModel.put(PACKAGE, DataModelMapper.getModelPackageName(mappingContext));
-        dataModel.put(CLASS_NAME, dataModelMapper.getModelClassNameWithPrefixAndSuffix(mappingContext, definition));
+        dataModel.put(CLASS_NAME, clazzName);
         dataModel.put(JAVA_DOC, definition.getJavaDoc());
         dataModel.put(ANNOTATIONS, graphQLTypeMapper.getAnnotations(mappingContext, definition));
-        dataModel.put(FIELDS, fieldDefinitionToParameterMapper.mapFields(
-                mappingContext, definition.getFieldDefinitions(), definition));
+        dataModel.put(FIELDS, fields);
         dataModel.put(GENERATED_INFO, mappingContext.getGeneratedInformation());
         dataModel.put(ENUM_IMPORT_IT_SELF_IN_SCALA, mappingContext.getEnumImportItSelfInScala());
         dataModel.put(IMMUTABLE_MODELS, mappingContext.getGenerateImmutableModels());
+        if (mappingContext.getParentInterfacePropertiesInKotlin().containsKey(clazzName)) {
```

Comment:
This is the only thing I have doubts about. Any ideas on how to improve it so that we can execute mappers independently?
Maybe (just for Kotlin) we can precompute these values beforehand? (In the similar way how I did in `ExtendedDocument.getTypesUnionsInterfacesNames()`)

Reply:
Does the document contain all the content? If so, it can be obtained directly from the document, and scala's `enumimportitself` in scala can do the same